crates/rpc/src/server/api.rs

```diff
@@ -88,6 +91,54 @@ impl RpcService {
 
         Self { store, block_producer }
     }
+
+    /// Fetches the genesis block header from the store.
+    ///
+    /// Automatically retries until the store connection becomes available.
+    pub async fn get_genesis_header_with_retry(&self) -> anyhow::Result<BlockHeader> {
+        let mut retry_counter = 0;
+        loop {
```

Comment:
Looks like we are now doing this pattern 3x in the code. Wondering if we could implement a utility fn that runs a closure with retries.

Reply:
#1072 contains some of my thoughts on this; but yeah its something I'm hoping will fall out of all the gRPC refactoring.